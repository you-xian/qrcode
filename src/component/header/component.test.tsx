import {
  AppBar,
  IconButton,
  Toolbar,
  Typography,
} from "@material-ui/core";
import HomeIcon from "@material-ui/icons/Home";
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import Header from "./index";

describe('<Header />', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<Header />)
  })

  test('It should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('It should mount following components', () => {
    expect(wrapper.find(AppBar)).toHaveLength(1);
    expect(wrapper.find(Toolbar)).toHaveLength(1);
    expect(wrapper.find(Typography)).toHaveLength(1);
    expect(wrapper.find(IconButton)).toHaveLength(1);
    expect(wrapper.find(HomeIcon)).toHaveLength(1);
  })

  test('renders an Typography .title', () => {
    expect(wrapper.find(Typography).hasClass(/makeStyles-title-\d+/)).toBeTruthy();
  });

  test('renders an HomeIcon .icon', () => {
    expect(wrapper.find(HomeIcon).hasClass(/makeStyles-icon-\d+/)).toBeTruthy();
  });

  test('It should click home button', () => {
    wrapper.find(IconButton).simulate("click");
    wrapper.find(IconButton).prop("onClick");
  })
})
