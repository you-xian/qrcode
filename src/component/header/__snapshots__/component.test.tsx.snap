// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> It should match snapshot 1`] = `
<WithStyles(ForwardRef(AppBar))
  position="static"
>
  <WithStyles(ForwardRef(Toolbar))>
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-title-3"
      variant="h6"
    >
      <WithStyles(ForwardRef(IconButton))
        color="inherit"
      >
        <Memo(ForwardRef(HomeIcon))
          className="makeStyles-icon-1"
        />
      </WithStyles(ForwardRef(IconButton))>
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Toolbar))>
</WithStyles(ForwardRef(AppBar))>
`;
